x-pack/solutions/security/plugins/security_solution/server/usage/detections/rules/update_usage.ts

```diff
@@ -165,6 +165,26 @@ export const updateRuleUsage = (
         totalType: 'elastic_total',
       }),
     };
+
+    if (detectionRuleMetric.is_customized) {
+      updatedUsage = {
+        ...updatedUsage,
+        elastic_customized_total: updateTotalUsage({
+          detectionRuleMetric,
+          updatedUsage,
+          totalType: 'elastic_customized_total',
+        }),
+      };
+    } else {
+      updatedUsage = {
+        ...updatedUsage,
+        elastic_noncustomized_total: updateTotalUsage({
+          detectionRuleMetric,
+          updatedUsage,
+          totalType: 'elastic_noncustomized_total',
+        }),
+      };
+    }
```

Comment:
This function is already overwhelmed with `if`/`else` conditions. It looks like we could benefit from extracting as minimum the new code to separate functions.